src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -76,6 +79,7 @@ internal class SolutionProjectGenerator
         /// Version 4.0
         /// </summary>
         private readonly Version _version40 = new Version(4, 0);
+#endif
```

Comment:
>f [](start = 5, length = 1)

nit: style: For all of the #endifs you added, consider using a comment after the #endif to make it clear what #if is being closed. E.g.   // FEATURE_ASPNET_COMPILER